src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -146,9 +152,20 @@ public void ICollection_NonGeneric_SyncRoot_NonNull(int count)
 
         [Theory]
         [MemberData(nameof(ValidCollectionSizes))]
+        public void ICollection_NonGeneric_SyncRoot_Null(int count)
+        {
+            if (ICollection_NonGeneric_SupportsSyncRoot && ICollection_NonGeneric_HasNullSyncRoot)
+            {
+                ICollection collection = NonGenericICollectionFactory(count);
+                Assert.Null(collection.SyncRoot);
+            }
+        }
```

Comment:
Rather than introducing a new test, I think I'd prefer if they were a single test that verified the SyncRoot was what it was supposed to be, e.g.
```C#
public void ICollection_NonGeneric_SyncRoot_NullOrNonNull(int count)
{
    if (!ICollection_NonGeneric_SupportsSyncRoot) return;

    ICollection collection = NonGenericICollectionFactory(count);
    Assert.Equal(ICollection_NonGeneric_HasNullSyncRoot, collection.SyncRoot == null);
}
```